src/lib/SoftHSM.cpp

```diff
@@ -362,8 +362,13 @@ CK_RV SoftHSM::C_Initialize(CK_VOID_PTR pInitArgs)
 		// Must be set to NULL_PTR in this version of PKCS#11
 		if (args->pReserved != NULL_PTR)
 		{
-			DEBUG_MSG("pReserved must be set to NULL_PTR");
-			return CKR_ARGUMENTS_BAD;
+			char *init_args = (char *)args->pReserved;
+
```

Comment:
We should also check that the length of the incoming string is larger than the length of "SOFTHSM2_CONF=". Just so that we at least get a single character when calling setConfigPath().